packages/client/hmi-client/src/workflow/workflow.ts

```diff
@@ -0,0 +1,72 @@
+// Defines the type of data an operation can consume and output
+export interface OperationData {
+	type: string;
+	label?: string;
+}
+
+// Defines a function: eg: model, simulate, calibrate
+export interface Operation {
+	name: string;
+	description: string;
+	action: Function;
+	validation?: Function;
+
+	inputs: OperationData[];
+	outputs: OperationData[];
+}
+
+// Defines the data-exchange between WorkflowNode
+// In most cases the value here will be an assetId
+export interface WorkflowPort {
+	type: string;
+	value?: any;
+}
```

Comment:
an `id` would be useful, just to name them.
```js
inputs: [{ id: 'model-a', type: 'model' }, { id:'model-b', type: 'model' }],
```

Reply:
I will add it as an optional field now now, I thought about this before, but backed off ... I didn't think an id-scheme would be manageable given that the number of ports are dynamic at runtime, but will have to rethink it.